.github/workflows/ci.yml

```diff
@@ -82,27 +82,44 @@ jobs:
       matrix:
         os: ${{ fromJson(needs.setup-os-matrix.outputs.os) }}
         library: [ILGPU, ILGPU.Algorithms]
+        framework: [netcoreapp3.1, net5.0]
+        include:
+          - os: windows-latest
+            library: [ILGPU, ILGPU.Algorithms]
+            framework: net47
       fail-fast: false
     runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
         uses: actions/checkout@v2
-      - name: Setup .NET Core SDK
+
+      # Latest .NET SDK - v5.0.x
+      - name: Setup the latest .NET SDK
+        if: matrix.framework != 'netcoreapp3.1'
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.x
+      
+      # Latest .NET SDK - v3.1.x
+      - name: Setup the latest .NET SDK v3.1.x
+        if: matrix.framework == 'netcoreapp3.1'
         uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: 3.1.407
-      - name: Pin .NET Core SDK
-        run: dotnet new globaljson --sdk-version 3.1.407
-      - name: Release Build
-        run: dotnet build --configuration=Release -p:TreatWarningsAsErrors=true Src
-      - name: CPU Tests
+          dotnet-version: 3.1.x
+
+      - name: Build and test - CPU
         if: matrix.os != 'cuda'
-        run: dotnet test --logger GitHubActions --no-build --configuration=Release Src/${{ matrix.library }}.Tests.CPU
+        run: |
+          dotnet build --configuration=Release --framework=${{ matrix.framework }} -p:TreatWarningsAsErrors=true Src/${{ matrix.library }}
```

Comment:
Now that you removed `--no-build` for `dotnet test`, it will automatically build, so there is no need for the `dotnet build` command
```suggestion
```